### PR TITLE
Adds emag effect for magboots

### PR DESCRIPTION
**Pull request by ghost**
[content]

Once applied, really high amount of item slowdown is applied without any of the benefits of magpulse or not slipping, can only be fixed with a solder

:cl:
 * rscadd: Applying a cryptographic sequencer to magboots will now disrupt their mag pulse attraction system. Use a solder to fix

### CODE REVIEW

**Comment by BomberBro** [comment]
>cannot be reversed

Since magboots are both a finite and necessary resource, it might be good to have some sort of fix like being able to solder them back to normal

**Comment by ghost** [comment]
> > cannot be reversed
> 
> Since magboots are both a finite and necessary resource, it might be good to have some sort of fix like being able to solder them back to normal

Will probably add emagging them again to put them back to normal

**Comment by BomberBro** [comment]
But it's an emag, it break things. It shouldn't be a magic card that can also FIX items since that's not it's "lore" purpose

**Comment by ghost** [comment]
> But it's an emag, it break things. It shouldn't be a magic card that can also FIX items since that's not it's "lore" purpose

Yeah I was thinking of the comms console, then I realised it's not fixed with an emag

**Comment by GreatBigFailure** [comment]
Make using a solder on them fix them. Poor solder should get more love.

**Comment by ghost** [comment]
> Make using a solder on them fix them. Poor solder should get more love.

This has been done

**Comment by GreatBigFailure** [comment]
That was fast. Almost like you had the idea before I posted that. I will still claim credit, you're welcome.

**Comment by ghost** [comment]
> That was fast. Almost like you had the idea before I posted that. I will still claim credit, you're welcome.

Earlier in the convo:

> > cannot be reversed
> 
> Since magboots are both a finite and necessary resource, it might be good to have some sort of fix like being able to solder them back to normal

**Comment by Eneocho** [comment]
Will the emag then fix break the CE's Magboots?

**Review by Kurfursten — DISMISSED** [review]
The way you've implemented this, it can be "repaired" by just turning the shoes off and on again.

In fact, if you emag magboots that are turned off, no one would even notice because turning them on would re-adjust the speed and flags.

The solution you have here of "you just can't turn off magboots that are emagged" is an unsatisfying one. You'll never trick anyone like that.

**Comment by PavelCIA** [comment]
make them unable to be turned off or taken off so people have to awkwardly shuffle to the nearest soldering iron at diona speed

**Comment by smalIboss** [comment]
> 
> 
> make them unable to be turned off or taken off so people have to awkwardly shuffle to the nearest soldering iron at diona speed

this is a really good idea
they're stuck to the ground so you can't take them off, it makes sense

**Comment by Exxion** [comment]
@Kurfursten If you can't turn them off, you can't fix them by turning them off. You can dislike the approach but it's not a code issue.

File: code/modules/clothing/shoes/magboots.dm [hunk 1]
```diff
@@ -96,9 +97,33 @@
 	..()
 	return
 
+/obj/item/clothing/shoes/magboots/attackby(var/obj/item/O, var/mob/user)
+	..()
+	if(isEmag(O))
```
Comment (ShiftyRail) [inline code/modules/clothing/shoes/magboots.dm]: Replace this snippet by `emag_act`

**Comment by ShiftyRail** [comment]
As Harris suggested, it's a finite resource, but to make it fixable with nothing more than a welder cheapens it a bit.

File: code/modules/clothing/shoes/magboots.dm (hunk 1 above)
Comment (ghost) [reply code/modules/clothing/shoes/magboots.dm]: Done

**Comment by ghost** [comment]
> Will the emag then fix break the CE's Magboots?

Magboots slowdown is stored in mag_slow, seperate from the general slowdown variable of clothing, (set in the emagging) so nothing is lost.

**Comment by Exxion** [comment]
Though I don't think it's worth blocking the PR over as said above, I do agree with Kurf that this isn't a great method. It would be better if the new behavior were applied in the proc that toggles the magboots instead of only being applied once when you emag them. That way they could be turned off and on just fine and keep the behavior.

File: code/modules/clothing/shoes/magboots.dm [hunk 2]
```diff
@@ -96,9 +97,34 @@
...
+			slowdown = NO_SLOWDOWN
+			icon_state = "[base_state]0"
+			to_chat(user, "<span class='notice'>You restore the mag-pulse traction system.</span>")
+			user.update_inv_shoes()	//so our mob-overlays update
+
 /obj/item/clothing/shoes/magboots/togglemagpulse(var/mob/user = usr)
 	if(user.isUnconscious())
 		return
+	if(emagged)
+		to_chat(user, "<span class='warning'>The mag-pulse traction system cannot be turned off!</span>")
```
Comment (jknpj) [inline code/modules/clothing/shoes/magboots.dm:126]: cannot be turned off?

**Comment by Dacendeth** [comment]
Perhaps you should just be able to order more magboots?

**Comment by Eneocho** [comment]
> 
> 
> Perhaps you should just be able to order more magboots?

A spacesuit kit. Has a normal spacesuit helmet, suit, breathing mask and magboots. Costs 150c.

**Comment by Eneocho** [comment]
Since this changes the base magboot behaviour, does this affect the Syndie Magboots or the Elite Syndie Magboots?
If it does, it should either not have an effect on syndie magboots or have a positive effect due to some syndicate tech synergy.
@kane-f

**Comment by TGFH** [comment]
Emagcreep. It's already one of the best possible purchases because it interacts with almost everything and has infinite uses. I don't really think it needs more.

**Comment by ghost** [comment]
> Since this changes the base magboot behaviour, does this affect the Syndie Magboots or the Elite Syndie Magboots?
> If it does, it should either not have an effect on syndie magboots or have a positive effect due to some syndicate tech synergy.
> @kane-f

I just made them immune now

**Comment by KingofHONK** [comment]
> Emagcreep. It's already one of the best possible purchases because it interacts with almost everything and has infinite uses. I don't really think it needs more.

Counterpoint: Emags are a staple traitor item. There is nothing wrong with traitors having powerful items/abilities.

**Comment by TGFH** [comment]
> 
> 
> > Emagcreep. It's already one of the best possible purchases because it interacts with almost everything and has infinite uses. I don't really think it needs more.
> 
> Counterpoint: Emags are a staple traitor item. There is nothing wrong with traitors having powerful items/abilities.

No, there's nothing wrong with that. I think it's an issue because if you make one item too good it just becomes a waste of TC to buy anything else.

**Comment by Theophrastus1** [comment]
> 
> 
> > > Emagcreep. It's already one of the best possible purchases because it interacts with almost everything and has infinite uses. I don't really think it needs more.
> > 
> > 
> > Counterpoint: Emags are a staple traitor item. There is nothing wrong with traitors having powerful items/abilities.
> 
> No, there's nothing wrong with that. I think it's an issue because if you make one item too good it just becomes a waste of TC to buy anything else.

Maybe if you're a powergaming tryhard for ebin valids. I'm a big fan of some of the lesser purchased items because you can do some fun gimmicks with it. If you just buy shit like emag revolver or ebow every traitor round, i don't think those items are the issue

**Comment by Kurfursten** [comment]
Compile failure

**Comment by ghost** [comment]
> Compile failure

Fixed

**Review by ShiftyRail — APPROVED** [review]

**Pull request merged**